fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneMapQueryTest.java

```diff
@@ -0,0 +1,461 @@
+                    .setGroup(i % 2)
+                    .build()
+            )
+            .collect(Collectors.toList());
+
+
+    private ExecutorService executorService = null;
+
+    @Override
+    public void setupPlanner(@Nullable PlannableIndexTypes indexTypes) {
+        if (indexTypes == null) {
+            indexTypes = new PlannableIndexTypes(
+                    Sets.newHashSet(IndexTypes.VALUE, IndexTypes.VERSION),
+                    Sets.newHashSet(IndexTypes.RANK, IndexTypes.TIME_WINDOW_LEADERBOARD),
+                    Sets.newHashSet(IndexTypes.TEXT),
+                    Sets.newHashSet(LuceneIndexTypes.LUCENE)
+            );
+        }
+        planner = new LucenePlanner(recordStore.getRecordMetaData(), recordStore.getRecordStoreState(), indexTypes, recordStore.getTimer());
+    }
+
+    @Override
+    protected RecordLayerPropertyStorage.Builder addDefaultProps(final RecordLayerPropertyStorage.Builder props) {
+        return super.addDefaultProps(props)
+                .addProp(LuceneRecordContextProperties.LUCENE_EXECUTOR_SERVICE, (Supplier<ExecutorService>)() -> executorService);
+    }
+
+    protected void openRecordStore(FDBRecordContext context) {
+        openRecordStore(context, md -> {
+        }, SIMPLE_TEXT_SUFFIXES);
+    }
+
+    protected void openRecordStore(FDBRecordContext context, RecordMetaDataHook hook, Index simpleDocIndex) {
+        RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecordsTextProto.getDescriptor());
+        metaDataBuilder.getRecordType(TextIndexTestUtils.COMPLEX_DOC).setPrimaryKey(concatenateFields("group", "doc_id"));
+        if (simpleDocIndex != null) {
+            metaDataBuilder.removeIndex("SimpleDocument$text");
+            metaDataBuilder.addIndex(TextIndexTestUtils.SIMPLE_DOC, simpleDocIndex);
+        }
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_LONG_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_WRAPPER_TO_LONG_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_LONG_WRAPPER_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_INT_LUCENE_INDEX);
+        metaDataBuilder.addIndex(MAP_DOC, MAP_STRING_2_DOUBLE_LUCENE_INDEX);
+        hook.apply(metaDataBuilder);
+        recordStore = getStoreBuilder(context, metaDataBuilder.getRecordMetaData())
+                .setSerializer(TextIndexTestUtils.COMPRESSING_SERIALIZER)
+                .createOrOpen();
+        setupPlanner(null);
+    }
+
+    private void initializeNested() {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context);
+            mapDocuments.forEach(recordStore::saveRecord);
+            commit(context);
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "d, true",
+            "Blah, false",
+            "c, false"
+    })
+    void mapStringToLong(String value, boolean found) throws Exception {
+        initializeNested();
```

Comment:
It seems like it would be good to have a document that has a different key for all of the entries (so it never matches)
Also, a document that doesn't have any entries of any time, which also would never match.

Reply:
Done.